### PR TITLE
Print Echo parentheses fixer

### PR DESCRIPTION
**Pull request by Soullivaneuh**
#1539 for echo/print.

I don't know if this is related to any PSR. I put it on Contrib for the moment.

### CODE REVIEW

**Comment by gharlan** [comment]
test case: `echo (1 + 2) . $foo;`

**Comment by Soullivaneuh** [comment]
@gharlan Will add it. For information:

```php
$foo = 'bar';

echo (1 + 2) . $foo."\n";
echo 1 + 2 . $foo."\n";
```

Output is the same, so we can remove parentheses. But is that really wanted?

I mean, parentheses of the complete argument should be removed. I don't think the others is the job of this fixer.

Furthermore, you can have some more tricky edge cases:

```php
echo (1 + 2) * 10, "\n"; // Output "30"
echo 1 + 2 * 10, "\n";   // Output "21"
```

**Comment by gharlan** [comment]
@Soullivaneuh I agree, I even didn't know that the result would be the same (in my test case without parantheses) :D

So it was meant as an example which should not be changed by this fixer.

**Comment by gharlan** [comment]
More test cases:

```php
echo ($a ? $b : $c); // should be changed
echo ($a ? $b : $c) ? $d : $e; // should not be changed
echo 10 * (2 + 3); // should not be changed
echo ('foo'), ('bar'); // i'm not sure
```

**Comment by Soullivaneuh** [comment]
```
echo ('foo'), ('bar'); // i'm not sure
```

@gharlan For me, this should not be changed. The signature does not have parentheses, this is OK.

**Comment by Soullivaneuh** [comment]
PR ready for review and merge.

**Comment by keradus** [comment]
fix the project itself

File: Symfony/CS/Fixer/Contrib/PrintEchoParenthesesFixer.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+class PrintEchoParenthesesFixer extends AbstractFixer
```
Comment (keradus) [inline Symfony/CS/Fixer/Contrib/PrintEchoParenthesesFixer.php]: final

File: Symfony/CS/Tests/Fixer/Contrib/PrintEchoParenthesesFixerTest.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Symfony\CS\Tests\Fixer\Contrib;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+class PrintEchoParenthesesFixerTest extends AbstractFixerTestBase
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/PrintEchoParenthesesFixerTest.php]: final, internal

**Comment by Soullivaneuh** [comment]
> fix the project itself

Ok. Do you think this fixer should be on any PSR? Or Symfony rules?

File: Symfony/CS/Tests/Fixer/Contrib/PrintEchoParenthesesFixerTest.php (hunk 2 above)
Comment (Soullivaneuh) [inline Symfony/CS/Tests/Fixer/Contrib/PrintEchoParenthesesFixerTest.php]: One day, I will remember...

**Comment by keradus** [comment]
Closing in favour of #1550

**Pull request closed without merging**